docs/source/user/jit-module.rst

```diff
@@ -0,0 +1,103 @@
+.. _jit-module:
+
+============================================
+Automatic module jitting with ``jit_module``
+============================================
+
+A common usage pattern is to have an entire module containing user-defined
+functions that all need to be jitted. One option to accomplish this is to
+manually apply the ``@jit`` decorator to each function definition. This approach
+works and is great in many cases. However, for large modules with many functions,
+manually ``jit``-wrapping each function definition can be tedious. For these
+situations, Numba provides another option, the ``jit_module`` function, to
+automatically replace functions declared in a module with their ``jit``-wrapped
+equivalents.
+
+It's important to note the contexts under which ``jit_module`` will *not*
+impact a function:
+
+1. Functions which have already been wrapped with a Numba decorator (e.g.
+   ``jit``, ``vectorize``, ``cfunc``, etc.) are not impacted by ``jit_module``.
+
+2. Functions which are declared outside the module from which ``jit_module``
+   is called are not automatically ``jit``-wrapped.
+
+3. Function declarations which occur logically prior to calling ``jit_module``
```

Comment:
Isn't the inverse the case? Functions declared logically before a ``jit_module`` call are jitted?